Create permissions utils

OmerLevy1248 opened the pull request.

OmerLevy1248 committed:
Create PermissionUtils class + refactor utilities classes names

OmerLevy1248 committed:
Move permission and snackber functions to global utils

OmerLevy1248 committed:
Align "NotificationUtils" API to one function (notifyUser)

Yuvalkeidar commented on app/src/main/java/com/gusta/wakemehome/utilities/PermissionUtils.java:
```diff
@@ -0,0 +1,117 @@
+
+        // Request permissions from the user
+        boolean shouldProvideRationale =
+                ActivityCompat.shouldShowRequestPermissionRationale(activity,
```
Why did you add it to specific permission ?
it's easy to change it to generic with permission name?

OmerLevy1248 replied:
You are right I will change.
I thought it would be problematic for the "settings" link in the snackbar, but it won't (all permissions should send user to "app details" setting).

Yuvalkeidar commented on app/src/main/java/com/gusta/wakemehome/utilities/NotificationUtils.java:
```diff
@@ -8,24 +9,117 @@
+     */
+    public static void notifyUser(Activity activity, int mainTextStringId, int actionTextStringId,
+                                  View.OnClickListener listener) {
+        notifyUser(activity, activity.getString(mainTextStringId), null,
```
which activity will be send on service?

OmerLevy1248 replied:
A service can't ask for permissions as it can't show the dialog box (actually it can but it's bad user experience)

Yuvalkeidar commented on app/src/main/java/com/gusta/wakemehome/geofencing/GeofenceManager.java:
```diff
@@ -77,7 +71,8 @@
-    public GeofenceManager(ContextWrapper activity, LiveData<? extends List<? extends GeofenceEntry>> liveData) {
+    public GeofenceManager(ContextWrapper activity,
+                           LiveData<? extends List<? extends GeofenceEntry>> liveData) {
         mContextWrapper = activity;
```
same question about the activity..

OmerLevy1248 replied:
same answer :)

Yuvalkeidar commented on app/src/main/java/com/gusta/wakemehome/AlarmAdapter.java:
```diff
@@ -80,7 +80,7 @@ public void onBindViewHolder(@NonNull AlarmViewHolder holder, int position) {
         String location = alarmEntry.getLocation();
         String radius =
-                WakeMeHomeUnitsUtils.formatLength(mContext, alarmEntry.getRadius()) + " radius";
+                UnitsUtils.formatLength(mContext, alarmEntry.getRadius()) + " radius";
```
radius should set to const

OmerLevy1248 replied:
What do you mean?

Yuvalkeidar commented on app/src/main/java/com/gusta/wakemehome/geofencing/GeofenceManager.java:
```diff
@@ -276,102 +277,15 @@ private void handleError(Exception e) {
-        // Provide an additional rationale to the user. This would happen if the user denied the
-        // request previously, but didn't check the "Don't ask again" checkbox.
-        if (shouldProvideRationale) {
-            Log.i(TAG, "Displaying permission rationale to provide additional context.");
```
I think it should be added to `strings.xml`

OmerLevy1248 replied:
This is a debug log - if we start adding them to "strings.xml" it will just make it a huge file we can't find anything at (Plus, while debugging we would usually add/removes logs like that - it will just irritate us to put them in strings.xml)

Yuvalkeidar commented on app/src/main/java/com/gusta/wakemehome/geofencing/GeofenceManager.java:
```diff
@@ -276,102 +277,15 @@ private void handleError(Exception e) {
-                        }
-                    });
-        } else {
-            Log.i(TAG, "Requesting permission");
```
`strings.xml`

OmerLevy1248 replied:
Debug log

Yuvalkeidar commented on app/src/main/java/com/gusta/wakemehome/utilities/PermissionUtils.java:
```diff
@@ -0,0 +1,117 @@
+        // Provide an additional rationale to the user. This would happen if the user denied the
+        // request previously, but didn't check the "Don't ask again" checkbox.
+        if (shouldProvideRationale) {
+            Log.i(TAG, "Displaying permission rationale to provide additional context.");
```
`Strings.xml`

OmerLevy1248 replied:
Debug log

Yuvalkeidar commented on app/src/main/java/com/gusta/wakemehome/utilities/PermissionUtils.java:
```diff
@@ -0,0 +1,117 @@
+                        }
+                    });
+        } else {
+            Log.i(TAG, "Requesting permission");
```
`String.xml`

OmerLevy1248 replied:
Debug log

Yuvalkeidar commented on app/src/main/java/com/gusta/wakemehome/utilities/PermissionUtils.java:
```diff
@@ -0,0 +1,117 @@
+        if (grantResults.length <= 0) {
+            // If user interaction was interrupted, the permission request is cancelled and you
+            // receive empty arrays.
+            Log.i(TAG, "User interaction was cancelled.");
```
All the string into `Strings.xml`, it should be able to be translatable

OmerLevy1248 replied:
Debug log.
No need to translate log. Only things that user sees.

OmerLevy1248 committed:
Code review changes:
- Remove "radius" string from alarm list items
- Add "Location" to permission methods names
- Add "modifyGeofences" function to Geofences manager
- Change "permission denied" message string

OmerLevy1248 committed:
Code review changes:
- Remove "radius" string from alarm list items
- Add "Location" to permission methods names
- Add "modifyGeofences" function to Geofences manager
- Change "permission denied" message string

OmerLevy1248 committed:
Merge remote-tracking branch 'origin/create_permissions_utils' into create_permissions_utils

OmerLevy1248 committed:
Merge branch 'master' into create_permissions_utils

# Conflicts:
#	app/src/main/java/com/gusta/wakemehome/AlarmAdapter.java
#	app/src/main/java/com/gusta/wakemehome/MainActivity.java
#	app/src/main/res/values/strings.xml
